submissions/Siusarna/data-structures/server/controllers.js

```diff
@@ -0,0 +1,118 @@
+const responseSender = require('./httpResponce.js');
+
+function isJSON(req) {
+  return req.headers['content-type'] === 'application/json';
+}
+
+function showList(res, list) {
+  const allNodeFromList = list.show();
+  const jsonObjWithAllNode = JSON.stringify({ data: allNodeFromList });
+  responseSender.success(res, jsonObjWithAllNode);
+}
+
+function deleteElementsFromStack(res, stack) {
+  try {
+    const removedValue = stack.pop();
+    const jsonObjWithRemovedValue = JSON.stringify({ data: removedValue });
+    responseSender.success(res, jsonObjWithRemovedValue);
+  } catch (err) {
+    responseSender.error(res, err.message);
+  }
+}
+
+function isValidData(inputData) {
+  return typeof inputData === 'number' || typeof inputData === 'string';
+}
+
+function wrongTypeOfValue(res) {
+  responseSender.error(res, 'wrong type of input value');
+}
+
+function wrongTypeOfContent(res) {
+  responseSender.error(res, 'wrong type of content');
+}
+
+function insertInList(res, list, inputData) {
+  if (!isValidData(inputData.data)) {
+    wrongTypeOfValue(res);
+    return;
+  }
+  if (inputData.successor && isValidData(inputData.successor)) {
+    try {
+      list.insert(inputData.data, inputData.successor);
+    } catch (err) {
+      throw new Error(err.message);
+    }
+  } else if (!inputData.successor) {
+    list.insert(inputData.data);
+  } else {
+    wrongTypeOfValue(res);
+  }
+}
+
+function processOfInsertInList(res, list, inputData) {
+  try {
+    insertInList(res, list, inputData);
+    responseSender.success(res);
+  } catch (err) {
+    responseSender.error(err.message);
+  }
+}
+
+function removeFromList(res, list, inputData) {
+  try {
+    list.remove(inputData.data);
+    responseSender.success(res);
+  } catch (err) {
+    responseSender.error(res, err.message);
+  }
+  res.end();
+}
+
+function pushOnStack(res, stack, inputData) {
+  stack.push(inputData.data);
+  responseSender.success(res);
+}
+
+function notFound(res) {
+  responseSender.error(res, 'Url not found');
+}
+
+function readBody(req, cb) {
+  let body = '';
+  req.on('data', chunk => {
+    body += chunk.toString();
+  });
+  req.on('end', () => {
+    cb(body);
+  });
+}
+
+module.exports = (list, stack) => {
```

Comment:
Too many nested checks :) Let's try to reduce it because it is hard to read right now.

One of possible ways is to create functions like `makeStackResponse` and `makeListResponse` and then call it like `responseMakers(typeOfStructures)`

Reply:
It's normal if I give `inputDate` for `responseMakers` , `makeStackResponse` and `makeListResponse` ?

Reply:
Not sure what you mean. Maybe you can provide some code examples?

Reply:
I just want to find out if this is correct
`function makeStackResponse(method, stack, inputData, res) {
  if (method === 'DELET') {
    deleteElementsFromStack(res, stack);
  } else if (method === 'POST') {
    pushOnStack(res, stack, inputData);
  }
}

function makeListResponse(method, list, inputData, res) {
  if (method === 'GET') {
    showList(res, list);
  } else if (method === 'POST') {
    insertInList(res, list, inputData);
  } else if (method === 'DELETE') {
    removeFromList(res, list, inputData);
  }
}

function responseMakers(typeOfStructures, method, inputData, res, list, stack) {
  if (typeOfStructures === 'list') {
    return makeListResponse(method, list, inputData, res);
  } else if (typeOfStructures === 'stack') {
    return makeStackResponse(method, stack, inputData, res);
  } else {
    return notFound(res);
  }
}`

Reply:
Ok, I just will try to write function as I imagine and commit it